pkg/loki/config_wrapper.go

```diff
@@ -76,11 +77,15 @@ func (c *ConfigWrapper) ApplyDynamicConfig() cfg.Source {
 			r.QueryScheduler.UseSchedulerRing = true
 		}
 
-		applyPathPrefixDefaults(r, defaults)
-		if err := applyIngesterRingConfig(r, &defaults); err != nil {
+		applyPathPrefixDefaults(r, &defaults)
```

Comment:
Btw, since `applyPathPrefixDefaults` is being invoked before `applyDynamicRingConfigs`, the check `reflect.DeepEqual(r.CompactorConfig.CompactorRing, defaults.CompactorConfig.CompactorRing)` will always be false I think

Reply:
does `applyPathPrefixDefaults` modify the compactor's ring? I see it modifying the working directory, but I'm missing the impact on the ring?

Reply:
Sorry, I completely confused the function names :joy: I was referring to `applyTokensFilePath`